playbooks/templates/kube_config_oidc.yml.j2

```diff
@@ -0,0 +1,32 @@
+apiVersion: v1
+clusters:
+-   cluster:
+        certificate-authority-data: {{ ca_cert.stdout }}
+        server: https://{{ kube_oidc_apiserver_endpoint | default(kube_apiserver_access_address) }}:{{ kube_oidc_apiserver_port | default(kube_apiserver_port) }}
+    name: {{ cluster_name }}
+contexts:
+-   context:
+        cluster: {{ cluster_name }}
+        namespace: default
+        user: admin@{{ cluster_name }}
+    name: {{ cluster_name }}
+current-context: {{ cluster_name }}
+kind: Config
+preferences: {}
+users:
+-   name: admin@{{ cluster_name }}
+    user:
+        exec:
+            apiVersion: client.authentication.k8s.io/v1beta1
+            args:
+            - oidc-login
+            - get-token
+            - --oidc-issuer-url={{ kube_oidc_url }}
+            - --oidc-client-id={{ kube_oidc_client_id }}
+            - --oidc-client-secret={{ kube_oidc_client_secret }}
+{% for scope in kube_oidc_extra_scopes %}
+            - --oidc-extra-scope={{ scope }}
```

Comment:
Don't we always want to add the scopes `kube_oidc_username_claim` and `kube_oidc_groups_claim`?

Reply:
I was thinking that it was more flexible this way, not sure if everyone always wants those two.

Reply:
If you are using those claim you need to include them in the scope otherwise they will not be part of the token. 

I guess that you could set it so if you set `kube_oidc_extra_scopes` you'll need to include them in the list so then you have a choice. But by default I think it's fine to always include both of them. Otherwise I feel like you'll always need to set them on both places and it risks being a source of failure.

Reply:
Forgot that they were used in other places. Yes I now agree that we should add them by default. I'll wrap them in an `if defined` so if they are not set, then they are ignored.
Then `kube_oidc_extra_scopes` will be added on top of those two.
Does that sound good?

Reply:
Yes, that sounds good.